### DISCUSSION TITLE
Get message "HTTP Error 403: rate limit exceeded" when trying to install

### DISCUSSION THREAD

**Discussion by iamdobi** [kiali/kiali#4140] [Q&A] [answered]
**Describe the bug**
When I try to install kiali_cr via kiali-operator, the kiali pod is not generated, but it says "HTTP Error 403: rate limit exceeded".
I wonder the meaning of this message.
I guess it maybe related to rate limit with docker registry, am I right?
Is there any way to check detail logs or messages with CR itself? (pod is not even created T.T)

![image](https://user-images.githubusercontent.com/15792655/124091214-8cf5d900-da90-11eb-99f7-6c36fc4f1da5.png)

**Versions used**
Kiali: v1.22.1, v1.26.0
Istio: 1.16
Kubernetes flavour and version: v1.15.11

**To Reproduce**
Just execute below command, as written in official guide:
```sh
shell> helm install \
    --version 1.26.0 \
    --set cr.create=true \
    --set cr.namespace=istio-system \
    --namespace kiali-operator \
    --repo https://kiali.org/helm-charts \
    kiali-operator \
    kiali-operator

... (few seconds later)

shell> kubectl describe kiali kiali -n istio-system
(You can refer to above screenshot)
```
The kiali-operator itself is installed successfully, only kiali-cr is problematic.

**Expected behavior**
The kiali_cr executed well and kiali-server pod will be created.

**Comment by jmazzitelli** [comment, marked as answer]
Moving to Q&A Discussion, as this is not a bug.

You only provided a screenshot, not the full Kiali CR yaml or the operator pod logs, so I can't say definitively, but I'm 90% sure this is because you are using the defaults of Kiali v1.26 operator which (unless you explicitly tell it to do otherwise) will query github.com for the last release of Kiali and will install that. But your IP is getting rate-limited by github and thus the error.

Two things:
* We changed this behavior of the operator getting the last release from github because it is not the proper thing to do (the v1.26 operator will not be able to install the last release of Kiali (v1.36)).
* To get around this, in your Kiali CR set `spec.deployment.image_version` to `v1.26` and things should start to work.

If you still need help, please post the full yaml of the Kiali CR after the operator hits the error, and please post the operator pod logs (particularly, the errors that you see in there).

**Reply by jmazzitelli** [reply]
As for that first bullet point I mentioned above, see this for the reason why we changed behavior (and why you should not rely on the default behavior of the v1.26 operator): https://github.com/kiali/kiali/issues/3968

**Reply by jmazzitelli** [reply]
> * To get around this, in your Kiali CR set `spec.deployment.image_version` to `v1.26` and things should start to work.

And to be clear, based on how you installed the Kiali operator (i.e. via helm and having it install the Kiali CR for you), you will want to pass this additional option to your helm command:

`--set cr.spec.deployment.image_version=v1.26`

**Reply by iamdobi** [reply]
Thanks to your rapid response.

First, I'm sorry that I didn't know about existence of "discussion" menu, T.T.


It was caused by git api, as you said.
I can workaround it with your guide.
It worked like a charm!!! You saved my life.

And I also agree with your opinion about this behavior. 
So thank you for your proper comment! :)
